### DISCUSSION TITLE
Implementation of RF module CC1101 with ESP32

### DISCUSSION THREAD

**Discussion by radek-turecki** [arendst/Tasmota#22715] [Ideas]
Hello,

Is there any time perspective to implement the CC1101 RF module support into Tasmota software to read data from electric meters?

Currently I am using via SDR device + RTL, but frequency switching does not work well. In addition, the SDR tuner must be plugged into the computer, which limits the range and using any module with ESP and WiFi + C11011 would solve this problem very elegantly, especially since smart electric meters are becoming more and more popular.

Any chance of implementing CC1101 module support into Tasmota?

**Comment by sfromis** [comment]
It is not so much a question of "chances", but about who'd be deciding to take on the work needed to  support it, in the same way as lots of other drivers have been submitted by people having an interest in getting something supported.

This project seems to be doing "something" wit Berry and CC1101, but I've not looked into it.
https://github.com/andrew01144/Tasmota-SomfyRTS/

**Comment by radek-turecki** [comment]
I'm too weak as a programmer, but there are certainly many young and smart guys here who could take up the topic. 

I only did data decodingand for only one electricity meter, it may be useful for someone:
https://community.openhab.org/t/electricity-meter-wireless-m-bus-wm-bus-apator-otus-3/160870

But to cut off even unnecessary discussion of what is better - Tasmota or ESPHome, in the case of Tasmota I miss this part....

**Reply by barbudor** [reply]
The question is not so much about the cc1101 but is there a firmware for the cc1101 which can capture the messages from your meter ?
Then what is the protocol to communicate between this firmware and another MCU.
If yes, it's 'just' about implementing this protocol into an Tasmota energy driver, possibly in Berry on an ESP32

This will be simpler by someone having access to that meter for debugging

**Reply by radek-turecki** [reply]
CC1101 is only receiver, maybe somone smart guy from RTL could help to implement some drivers. They are implementing many devices and also energy meters are readable through RTL:

[https://github.com/merbanan/rtl_433
](url)
